src/main/java/build/buildfarm/instance/InstanceBase.java

```diff
@@ -0,0 +1,62 @@
+package build.buildfarm.instance;
+
+import java.util.UUID;
+
+public abstract class InstanceBase implements Instance {
+  private final String name;
+
+  protected static final String BINDING_EXECUTIONS = "executions";
+  protected static final String BINDING_TOOL_INVOCATIONS = "toolInvocations";
+  protected static final String BINDING_CORRELATED_INVOCATIONS = "correlatedInvocations";
+
+  public InstanceBase(String name) {
+    this.name = name;
+  }
+
+  @Override
+  public String getName() {
+    return name;
+  }
+
+  private String bind(String binding, UUID id) {
+    String name = getName() + "/" + binding;
+    if (name.isEmpty()) {
+      return name;
+    }
```

Comment:
bug: name can never be empty as it's will have a "/" in it

Reply:
I really have no idea what I was thinking with this - probably got stuck in there from an earlier version of string mutation collapse. I'll just return the single line concat here.